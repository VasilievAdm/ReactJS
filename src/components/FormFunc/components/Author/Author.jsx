import React from 'react';

export const Author = (props) => {
  return (
    <input
      className="author"
      type="text"
      value={props.author}
      onChange={props.change}
      placeholder="Who are you?"
    />
  );
};
