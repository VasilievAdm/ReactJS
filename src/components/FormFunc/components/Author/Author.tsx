import React, { FC } from 'react';

interface AuthorProps {
  change: (e: unknown) => void;
  author: string;
}

export const Author: FC<AuthorProps> = (props) => {
  return (
    <input
      className="author"
      type="text"
      value={props.author}
      onChange={props.change}
      placeholder="Who are you?"
    />
  );
};
