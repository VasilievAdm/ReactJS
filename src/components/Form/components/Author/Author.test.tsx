import React from 'react';
import { Author } from './Author';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Author', () => {
  it('render component', () => {
    render(<Author value="" change={() => void {}} />);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<Author value="" change={() => void {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render component with placeholder', () => {
    render(<Author value="" change={() => void {}} />);
    expect(screen.getByPlaceholderText(/Who are you?/)).toBeInTheDocument();
  });

  it('render component without error', () => {
    render(<Author value="" change={() => void {}} />);
    expect(screen.queryByText('Error')).not.toBeInTheDocument();
  });

  it('render multiply components', () => {
    render(
      <>
        <Author value="" change={() => void {}} />
        <Author value="" change={() => void {}} />
      </>
    );

    expect(screen.queryAllByRole('textbox').length).toBe(2);
  });

  it('render component with text', () => {
    render(<Author value="user" change={() => void {}} />);
    expect(screen.getByRole('textbox')).toHaveValue('user');
  });
});
