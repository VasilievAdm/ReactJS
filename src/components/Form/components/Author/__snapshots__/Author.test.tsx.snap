// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author render with snapshot 1`] = `
<DocumentFragment>
  <input
    class="author"
    placeholder="Who are you?"
    type="text"
    value=""
  />
</DocumentFragment>
`;
